OpenRA.Mods.D2k/Traits/VariedCostMultiplier.cs

```diff
@@ -0,0 +1,39 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
+{
+	[Desc("Modifies the production cost of this actor for a specific queue or when a prerequisite is granted. " +
+		"Requires `VariedCostManager` on the world actor.")]
+	public class VariedCostMultiplierInfo : TraitInfo<VariedCostMultiplier>, IProductionCostModifierInfo
+	{
+		[Desc("Only apply this cost change if owner has these prerequisites.")]
+		public readonly string[] Prerequisites = { };
+
+		[Desc("Queues that this cost will apply.")]
+		public readonly HashSet<string> Queue = new HashSet<string>();
+
+		int IProductionCostModifierInfo.GetProductionCostModifier(TechTree techTree, string queue, World world)
+		{
+			if ((!Queue.Any() || Queue.Contains(queue)) && (!Prerequisites.Any() || techTree.HasPrerequisites(Prerequisites)))
+				return world.WorldActor.Trait<VariedCostManager>().CostPercentage;
+
+			return 100;
+		}
```

Comment:
Can you add a IRulesetReloaded to check the existance of VariedCostManager on the world/player actor?

Reply:
Sure.